### PR TITLE
feat: add suggested collaborators support to UnifiedShareModal

### PR DESCRIPTION
**Pull request by nickgriffin1**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @nickgriffin1 has signed the CLA. Thanks for the pull request!

File: src/features/unified-share-modal/messages.js [hunk 1]
```diff
@@ -120,6 +120,11 @@ const messages = defineMessages({
         description: 'Label for "Message" text box to email the Shared Link',
         id: 'boxui.unifiedShare.message',
     },
+    suggestedCollabsTitle: {
+        defaultMessage: 'Suggested:',
```
Comment (priyajeet) [inline src/features/unified-share-modal/messages.js]: Should `:` be in the message? VS having it in the render()

File: src/features/unified-share-modal/ContactsField.js [hunk 2]
```diff
@@ -28,7 +28,9 @@ type Props = {
     onContactAdd: Function,
     onContactRemove: Function,
     onInput?: Function,
+    onSuggestedCollaboratorAdd: Function,
```
Comment (priyajeet) [inline src/features/unified-share-modal/ContactsField.js]: Can the `Function` be made specific? Like () => void or something

File: src/features/unified-share-modal/messages.js (hunk 1 above)
Comment (nickgriffin1) [reply src/features/unified-share-modal/messages.js]: I previously asked the Moji team about this. According to them, it is best to include colon in the translation.

File: src/features/unified-share-modal/ContactsField.js (hunk 2 above)
Comment (nickgriffin1) [reply src/features/unified-share-modal/ContactsField.js]: I'll make this more specific.

File: src/features/unified-share-modal/UnifiedShareModal.js [hunk 3]
```diff
@@ -90,6 +90,8 @@ type Props = {
      * This function should return a Promise.
      */
     sendSharedLinkError: React.Node,
+    /** Function hoists contact data upon updates to the parent component. Only needed for suggested collabs. */
+    setUpdatedContacts?: Function,
```
Comment (priyajeet) [inline src/features/unified-share-modal/UnifiedShareModal.js]: Can this be made more specific?

Comment (nickgriffin1) [reply src/features/unified-share-modal/UnifiedShareModal.js]: Sure I'll update this and get someone who has worked on this feature before to take a look.

**Pull request merged**